src/protocols/secure_channel/PASESession.cpp

```diff
@@ -139,10 +139,7 @@ CHIP_ERROR PASESession::GeneratePASEVerifier(Spake2pVerifier & verifier, uint32_
 
     if (useRandomPIN)
     {
-        ReturnErrorOnFailure(DRBG_get_bytes(reinterpret_cast<uint8_t *>(&setupPINCode), sizeof(setupPINCode)));
-
-        // Passcodes shall be restricted to the values 00000001 to 99999998 in decimal, see 5.1.1.6
-        setupPINCode = (setupPINCode % kSetupPINCodeMaximumValue) + 1;
+        SetupPayload::generateRandomSetupPin(setupPINCode);
```

Comment:
generateRandomSetupPin returns a CHIP_ERROR. We should handle that.

```suggestion
        ReturnErrorOnFailure(SetupPayload::generateRandomSetupPin(setupPINCode));
```